packages/material-ui/scripts/rollup.config.js

```diff
@@ -7,7 +7,6 @@ import { uglify } from 'rollup-plugin-uglify';
 import { sizeSnapshot } from 'rollup-plugin-size-snapshot';
 
 const input = './src/index.js';
-const name = 'material-ui';
```

Comment:
Could you please change only this line?

Reply:
If I keep and change `name` to be `'MaterialUI'` then the file name will be changed aswell: `build/umd/MaterialUI.production.min.js`. 
Probably not what we want, dashes are more common and better reflects the project name/typo: `build/umd/material-ui.production.min.js`.

Reply:
Okay, hardcode also file names but reuse the `name`.

Reply:
I didn't do it on purpose because it does not add any value.

If you variabilize `name` then the file name should be treated equally:
```JS
const baseFileName = 'build/umd/material-ui';
const outputName = 'MaterialUI';
// ...
output: {
  file: `${baseFileName}.development.js`,
  format: 'umd',
  name: outputName,
  globals
}
```

vs

```JS
// ...
output: {
  file: 'build/umd/material-ui.development.js',
  format: 'umd',
  name: 'MaterialUI',
  globals
}
```

I believe the later form is more readable (the file is only 60 lines long and will stay small over time).
Sometimes variabilize/factorize small things can make them less readable.

<br><br>

We can try to be a smart asses and reduce lines of code by 2:
```JS
const getConfig = mode => ({
  input,
  output: {
    file: `build/umd/material-ui.${mode}.js`,
    format: 'umd',
    name: 'MaterialUI',
    globals
  },
  external: Object.keys(globals),
  plugins: [
    nodeResolve(),
    babel(babelOptions),
    commonjs(commonjsOptions),
    nodeGlobals(),
    replace({ 'process.env.NODE_ENV': JSON.stringify(mode) }),
    mode === 'production' && sizeSnapshot(),
    mode === 'production' && uglify()
  ]
});

export default [
  getConfig('development'),
  getConfig('production')
];
```
well, the added value on a very small file like this one is 0 and just make things harder IMHO.

Reply:
I mean hard code file name in rollup config. Without such tricky overconfiguration. And reuse the `name` of global variable.

Reply:
```
import nodeResolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import babel from 'rollup-plugin-babel';
import replace from 'rollup-plugin-replace';
import nodeGlobals from 'rollup-plugin-node-globals';
import { uglify } from 'rollup-plugin-uglify';
import { sizeSnapshot } from 'rollup-plugin-size-snapshot';

const input = './src/index.js';
const name = 'MaterialUI';
const globals = {
  react: 'React',
  'react-dom': 'ReactDOM',
};
const babelOptions = {
  exclude: /node_modules/,
  // We are using @babel/plugin-transform-runtime
  runtimeHelpers: true,
  configFile: '../../babel.config.js',
};
const commonjsOptions = {
  ignoreGlobal: true,
  include: /node_modules/,
};

export default [
  {
    input,
    output: {
      file: 'build/umd/material-ui.development.js',
      format: 'umd',
      name,
      globals,
    },
    external: Object.keys(globals),
    plugins: [
      nodeResolve(),
      babel(babelOptions),
      commonjs(commonjsOptions),
      nodeGlobals(),
      replace({ 'process.env.NODE_ENV': JSON.stringify('development') }),
    ],
  },
  {
    input,
    output: {
      file: 'build/umd/material-ui.production.min.js',
      format: 'umd',
      name,
      globals,
    },
    external: Object.keys(globals),
    plugins: [
      nodeResolve(),
      babel(babelOptions),
      commonjs(commonjsOptions),
      nodeGlobals(),
      replace({ 'process.env.NODE_ENV': JSON.stringify('production') }),
      sizeSnapshot(),
      uglify(),
    ],
  },
];
```